src/apps/external_executor/protobuf/executor_registration.proto

```diff
@@ -61,28 +61,10 @@ message NewExecutor
   repeated EndpointKey supported_endpoints = 3;
 }
 
-// Responsse indicating whether a NewExecutor was accepted
+// Response indicating whether a NewExecutor was accepted
 message RegistrationResult
 {
-  enum Outcome {
-    // The attestation was fully verified, and this executor is now registered
-    ACCEPTED = 0;
-
-    // Unable to verify attestation
-    ATTESTATION_VERIFICATION_FAILURE = 1;
-
-    // Measurement described by attestation is not trusted
-    MEASUREMENT_NOT_TRUSTED = 2;
-
-    // The given public cert is already registered on this node for a
-    // distinct code ID. Note that calls should be idempotent, so repeated
-    // registration with identical requests should return ACCEPTED
-    DUPLICATE_CERT = 3;
-
-    // Any error not covered above
-    REJECTED = 4;
-  }
-  Outcome outcome = 1;
```

Comment:
These are covered by gRPC error responses `ccf::grpc::make_error`

Reply:
If we're going this route, I think we should remove the `outcome` field entirely. It's not helpful to give them an additional string to parse, it should either be an enum with limited values, or a description for humans (the latter is already provided by `details`). We should probably insert the `executor_id` in here as well - the Executor never _needs_ to know this ID we generated for them, but it may help correlate things in debugging logs.

Separately, I'm not sure if switching from in-body error messages to gRPC errors is a good idea, as there's a risk of the generated code being awkward for clients to use. In Python for instance, this will result in a bunch of try-catches around registration calls. I think it's helpful to distinguish fundamental errors (you didn't create the right kind of channel, you didn't format your body correctly, you didn't send the correct argument type) which are gRPC errors, from application-space errors (basically as soon as you've given us a parseable `NewExecutor` message) which can be reported in application-space response bodies. This is maybe an unnecessary distinction - we don't do it for HTTP, where both most of our apps and the framework report failures by the status code. But we should work out what's the right pattern for gRPC.

Reply:
Ok, ignore the second part.

After a little searching, it looks like standard gRPC practice is to use errors at the gRPC level (ie - `grpc-status` and associated details) rather than in app-space. This gets converted to client side exceptions in languages that don't mind exceptions, and manual failure-checking in things like C++. That seems backwards to me - you can't do the same for streamed responses, so you have a weird step-change in behaviour when you switch from unary response (might throw) to streamed response (check the body of each response for errors, a throw means a channel-level failure). But I don't make the precedent.